frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -0,0 +1,436 @@
+import * as _ from 'lodash-es';
+import { LabelSelector } from '@console/internal/module/k8s/label-selector';
+import { getRouteWebURL } from '@console/internal/components/routes';
+import { TopologyDataResources, ResourceProps, TopologyDataModel } from './topology-types';
+
+export const podColor = {
+  Running: '#0066CC',
+  'Not Ready': '#519DE9',
+  Warning: '#F0AB00',
+  Empty: '#FFFFFF',
+  Failed: '#CC0000',
+  Pending: '#8BC1F7',
+  Succceeded: '#519149',
+  Terminating: '#002F5D',
+  Unknown: '#A18FFF',
+  'Scaled To 0': '#FFFFFF',
+};
+
+export const podStatus = Object.keys(podColor);
+
+function numContainersReadyFilter(pod) {
+  let numReady = 0;
+  _.forEach(pod.status.containerStatuses, function(status) {
+    if (status.ready) {
+      numReady++;
+    }
+  });
+  return numReady;
+}
+
+function isReady(pod) {
+  const numReady = numContainersReadyFilter(pod);
+  const total = _.size(pod.spec.containers);
+
+  return numReady === total;
+}
+
+function isContainerFailedFilter(containerStatus) {
+  return containerStatus.state.terminated && containerStatus.state.terminated.exitCode !== 0;
+}
+
+function isContainerLoopingFilter(containerStatus) {
+  return (
+    containerStatus.state.waiting && containerStatus.state.waiting.reason === 'CrashLoopBackOff'
+  );
+}
+
+function podWarnings(pod) {
```

Comment:
We should line up with what the project status page does here. Note a lot of this looks like it was taken from old code written before pods had container statuses (pre k8s 1.0). 

Let's align with the simpler logic here:

https://github.com/openshift/console/blob/master/frontend/public/components/overview/index.tsx#L150-L178

Reply:
@spadgett is this something we can take up in a bug later? I'll make note and add it to my growing collection :)

Reply:
> @spadgett is this something we can take up in a bug later? I'll make note and add it to my growing collection :)

Works for me